.devcontainer/Dockerfile

```diff
@@ -1,22 +1,27 @@
 FROM mcr.microsoft.com/dotnet/sdk:3.1-focal
 
-# Install Mono and Nuget CLI
-RUN apt-get -y update && \
-    apt-get -y install dirmngr gnupg apt-transport-https ca-certificates && \
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF && \
-    sh -c 'echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" > /etc/apt/sources.list.d/mono-official-stable.list' && \
-    apt-get -y update && \
-    apt-get -y install --no-install-recommends mono-complete nuget && \
-    apt-get clean && rm -rf /var/lib/apt/lists/
-
-RUN apt update \
-    && apt-get install -y cmake \
+# Install Mono, NuGet, CMake, Ninja, Clang-11, build-essential
+RUN apt-get -y update \
+    && apt-get -y install dirmngr gnupg apt-transport-https ca-certificates \
+    && apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF \
+    && sh -c 'echo "deb https://download.mono-project.com/repo/ubuntu stable-focal main" > /etc/apt/sources.list.d/mono-official-stable.list' \
+    && apt-get -y update \
+    && apt-get -y install --no-install-recommends mono-complete nuget \
+    && apt-get -y install cmake \
                           ninja-build \
                           clang-11 \
                           clang-tidy-11 \
                           build-essential \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/
+    && apt-get clean && rm -rf /var/lib/apt/lists/
+
+# Install CMake 3.20 (required since apt-get uses 3.16 and repo requires 3.20)
+RUN curl -SsL https://github.com/Kitware/CMake/releases/download/v3.20.5/cmake-3.20.5-linux-x86_64.sh -o cmakeinstall.sh \
+    && echo "f582e02696ceee81818dc3378531804b2213ed41c2a8bc566253d16d894cefab cmakeinstall.sh" | sha256sum -c --strict - \
+    && chmod +x cmakeinstall.sh \
+    && ./cmakeinstall.sh --prefix=/usr/local --exclude-subdir \
+    && rm cmakeinstall.sh
 
+# Install Rust
 RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
```

Comment:
I don't think this will create a usable rust env for dev containers. See [rust-debian.sh](https://github.com/microsoft/qsharp-compiler/blob/26eded8e4938a4fc9b636ff01d523c12d7ac7f4f/src/QirTools/pyqir/.devcontainer/library-scripts/rust-debian.sh) and [Dockerfile](https://github.com/microsoft/qsharp-compiler/blob/26eded8e4938a4fc9b636ff01d523c12d7ac7f4f/src/QirTools/pyqir/.devcontainer/Dockerfile#L3-L26).

Reply:
I'm not sure I understand. This works to compile the experimental open system simulator in Rust, and seems sufficient for other related Rust projects I've done. Why would additional handling need to be done?

Reply:
To clarify, the entire runtime repo is built via the `build/build.ps1` script, after having initially executed `bootstrap.ps1` the first time. Creating this development container, running bootstrap, build and test scripts succeeds, including Rust components.

Reply:
@idavis Any concerns still remaining here? This installation mechanism has been working for all of my container dev environments, and was originally added by @cgranade when we first added Rust code to this repo.

Reply:
We landed at using the universal devcontainer approach as used by several Azure SDKs and other projects.